File: 백준/Bronze/15593. Lifeguards （Bronze）/Lifeguards （Bronze）.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	static StringTokenizer st;
	
	public static void main(String[] args) throws NumberFormatException, IOException {
		
		int N = Integer.parseInt(br.readLine());
		
		int[] timeTable = new int[1001];
		int[][] lifguards = new int[N][2];
		
		for (int i = 0; i < N; i++) {
			st = new StringTokenizer(br.readLine());
			int A = Integer.parseInt(st.nextToken());
			int B = Integer.parseInt(st.nextToken());
			lifguards[i][0] = A;
			lifguards[i][1] = B;
			for (int j = A; j < B; j++) {
				timeTable[j]++;
			}
		}
		
		int total = 0;
		for (int i = 0; i < 1001; i++) {
			if (timeTable[i] > 0)
				total++;
		}
		
		int answer = 0;
		for (int i = 0; i < N; i++) {
			int result = total;
			int a = lifguards[i][0];
			int b = lifguards[i][1];
			for (int j = a; j < b; j++) {
				if (timeTable[j] == 1)
					result--;
			}
			answer = Math.max(answer, result);
		}
		
		System.out.println(answer);
		
	} // end of main
} // end of class

File: 백준/Bronze/15593. Lifeguards （Bronze）/Lifeguards （Bronze）.py
"""
완전탐색


0부터 1000까지 일하는 라이프 가드 수를 모두 저장하고

소를 하나만 빼보기
(해당 라이프가드 업무시간에 일하는 소가 1이라면 이번 소가 빠지면 0이 된다는 뜻)


"""

import sys

input = lambda: sys.stdin.readline().strip()

time_table = [0] * 1001

N = int(input())

lifeguards = []

for _ in range(N):
    A, B = map(int, input().split())
    lifeguards.append((A, B))
    for i in range(A, B):
        time_table[i] += 1

answer = 0
total = 0

for i in range(1001):
    if time_table[i] > 0:
        total += 1

for i in range(N):
    result = total
    A, B = lifeguards[i]
    for j in range(A, B):
        if time_table[j] == 1:
            result -= 1
    answer = max(answer, result)

print(answer)